tools/column_maker/column_maker.xml

```diff
@@ -1,164 +1,170 @@
-            <output name="out_file1" file="column_maker_out3.interval"/>
-        </test>
-        <test>
-            <param name="cond" value="float(.0000000000001)"/>
-            <param name="input" value="1.bed"/>
-            <param name="round" value="no"/>
-            <output name="out_file1">
-                <assert_contents>
-                    <has_text text="CCDS10397" />
-                    <has_text text="1e-13" />
-                </assert_contents>
-            </output>
-        </test>
-        <test>
-            <param name="cond" value="float(.0000000000001)"/>
-            <param name="input" value="1.bed"/>
-            <param name="round" value="no"/>
-            <param name="avoid_scientific_notation" value="yes"/>
-            <output name="out_file1">
-                <assert_contents>
-                    <has_text text="CCDS10397" />
-                    <has_text text=".0000000000001" />
-                    <not_has_text text="1e-13" />
-                </assert_contents>
-            </output>
-        </test>
-    </tests>
-    <help>
-
- .. class:: infomark
-
-**TIP:** If your data is not TAB delimited, use *Text Manipulation-&gt;Convert*
-
------
-
-**What it does**
-
-This tool computes an expression for every row of a dataset and appends the result as a new column (field).
-
-- Columns are referenced with **c** and a **number**. For example, **c1** refers to the first column of a tab-delimited file
-
-- **c3-c2** will add a length column to the dataset if **c2** and **c3** are start and end position
-
------
-
-**Example**
-
-If this is your input::
-
-   chr1  151077881  151077918  2  200  -
-   chr1  151081985  151082078  3  500  +
-
-computing "c4*c5" will produce::
-
-   chr1  151077881  151077918  2  200  -   400.0
-   chr1  151081985  151082078  3  500  +  1500.0
-
-if, at the same time, "Round result?" is set to **YES** results will look like this::
-
-   chr1  151077881  151077918  2  200  -   400
-   chr1  151081985  151082078  3  500  +  1500
-
-You can also use this tool to evaluate expressions. For example, computing "c3>=c2" for Input will result in the following::
-
-   chr1  151077881  151077918  2  200  -  True
-   chr1  151081985  151082078  3  500  +  True
-
-or computing "type(c2)==type('') for Input will return::
-
-   chr1  151077881  151077918  2  200  -  False
-   chr1  151081985  151082078  3  500  +  False
-
-
-The following built-in functions are available::
-
-  abs | all | any | bin | bool | chr | ceil | cmp | complex
-
-  divmod | exp | float | log | log10 | floor | hex | int | len | long
-
-  max | min | oct | ord | pow | range | reversed
-
-  round | sorted | sqrt | str | sum | type | unichr | unicode |
-
-    </help>
-    <citations />
-</tool>
+<tool id="Add_a_column1" name="Compute" version="1.3.1">
+    <description>an expression on every row</description>
+    <requirements>
+        <requirement type="package" version="2.7.13">python</requirement>
```

Comment:
can we migrate this to python3

Reply:
seems that the tests work for py 3. 

but there might be functional differences, e.g. integer division now returns float. or is the code using numpy?

Reply:
I would bump the tool version and port it to python3 - it the results are different that would be fine because its a different version. We do need to make the cut at some point.